interp/interp.go

```diff
@@ -428,6 +426,7 @@ func (interp *Interpreter) Repl(in, out *os.File) {
 	prompt := getPrompt(in, out)
 	prompt()
 	src := ""
+	interp.opt.cancelChan = true
```

Comment:
Please don't do this. This makes cancellation outside the yeagi REPL method unavailable. This in turn means that other REPL, say using peterh/liner cannot cancel, and particularly important for me (i.e. why I'm working on this), that cancellation of `Eval` calls is not possible.

Reply:
Ok. An exported method on Interpreter object, or making `CancelChan` an exported field of interpreter would work for you then ?

Reply:
I'm not a fan of masses of knobs; too much time in the Go ecosystem and seeing people abuse them in other contexts.

The issue here is the choice of the default; I think that the fast option should not be the default since it leaves the unaltered interpreter (the value returned by `interp.New` with default options) in a state that leaks goroutines. This will be a constant source of issues for you and for client code that uses the interpretter embedded. Then what to call it? It needs to be a name that indicates both fast and unsafe (which it is).

My preferred option would be an environment variable (probably undocumented), `YAEGI_FAST_CHAN`, that people who want to use this can if they dig hard enough; it should not be encouraged.

Reply:
As a side note, no matter the value `cancelChan`, and the use of cancellable or blocking chan read / write, `EvalWithContext` will always return and work as expected for integration within a custom REPL. You already did everything for that. What changes is that some background goroutines may remain hanged on a chan in one case, or released in another, but no change from `EvalWithContext` point of view. Or am I missing something ?

Reply:
Yeah, I understand that. I'm concerned about the goroutine leaks and resource depletion attack vectors.